snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -1029,18 +1029,34 @@ def _expand_part_properties(part_properties, part_schema):
     return properties
 
 
+def _resolve_path_parents(relative_path: str, base_directory: str) -> str:
```

Comment:
How about moving this to fileutils (removing the _) and maybe calling it something closer to the docstring description (e.g.; `get_relative_path`)

I am almost certain we have an implementation for this somewhere in our code too, so this could _eventually_ replace that and its uses

Reply:
Good idea, will do.

Reply:
I went with get_resolved_relative_path